app/controllers/ManageGroupClientsController.scala

```diff
@@ -211,20 +211,15 @@ class ManageGroupClientsController @Inject()
     }
   }
 
-  // TODO needs new BE endpoint APB-6886 to replace withGroupForAuthorisedOptedAgent
-  // currently handles add + remove, simplify to add and have separate bulk remove
   def showManageGroupClients(groupId: String, page: Option[Int] = None, pageSize: Option[Int] = None): Action[AnyContent] = Action.async { implicit request =>
     withGroupForAuthorisedOptedAgent(groupId) { group =>
       withSessionItem[String](CLIENT_FILTER_INPUT) { clientFilterTerm =>
         withSessionItem[String](CLIENT_SEARCH_INPUT) { clientSearchTerm =>
+          val clientsInGroupAlready = group.clients.toSeq.flatten.map(DisplayClient.fromClient(_)).map(_.copy(selected = true))
           for {
-            _ <- sessionCacheService.get(SELECTED_CLIENTS).map(maybeSelectedClients =>
-              if (maybeSelectedClients.isEmpty) {
-                val clientsInGroupAlready = group.clients.toSeq.flatten.map(DisplayClient.fromClient(_)).map(_.copy(selected = true))
-                sessionCacheService.put[Seq[DisplayClient]](SELECTED_CLIENTS, clientsInGroupAlready)
-              }
-            )
-            paginatedClients <- clientService.getPaginatedClients(group.arn)(page.getOrElse(1), pageSize.getOrElse(20))
+            _ <- sessionCacheService.put[Seq[DisplayClient]](EXISTING_CLIENTS, clientsInGroupAlready)
```

Comment:
You should only need to put in `EXISTING_CLIENTS` the first time you hit this page, but clear it if you go back to "Manage Groups" or complete the transaction of adding someone. It'll do for now